frontend/src/components/TodoItem.tsx

```diff
@@ -15,59 +15,46 @@ export type TodoItemProps = {
 };
 
 function TodoItem(props: TodoItemProps) {
-  const [done, setDone] = useState(props.done);
-  const [isLoading, setIsLoading] = useState(false);
+  // const [done, setDone] = useState(props.done); Checkpoint 2: To uncomment.
 
-  const updateTodoItem = useCallback(async () => {
+  const updateTodoItem = useCallback(async () => { // Checkpoint 2: How can we pass the `done` state to our endpoint?
     await axios.put(`${CONFIG.API_ENDPOINT}/todos/${props.id}`, {
       id: props.id,
       description: props.description,
-      done: done,
+      done: false,
     });
-  }, [props.description, props.id, done]);
+  }, [props.description, props.id]);
 
-  const deleteTodoItem = useCallback(async () => {
-    await axios.delete(`${CONFIG.API_ENDPOINT}/todos/${props.id}`);
-    props.refreshToDos();
-  }, [props.id, props.refreshToDos]);
+  // Checkpoint 3: Implement a method like "updateTodoItem" to delete a TodoItem
 
-  const generateGptContent = useCallback(async (id: string) => {
-    setIsLoading(true);
-    await axios.post(`${CONFIG.API_ENDPOINT}/todos/${id}/generate`);
-    setIsLoading(false)
-    props.refreshToDos();
-  }, []);
+  // Checkpoint 4: Implement a method like "updateTodoItem" to generate the ChatGPT content
+  // Considering how long it takes to generate the content, how could using `setIsLoading` be helpful? 
```

Comment:
Just a suggestion: do you think we could split this into checkpoint 4.1 and 4.2? 
My thought process for the reasoning behind splitting is: Once they create the 'generateGptContent' function, we can get them to observe how there's no UI indication that the function is actually working at the backend. Then from here we can let them better understand the rationale of having isLoading. What are your thoughts?

Reply:
sgtm!